pkg/limits/frontend/ring_test.go

```diff
@@ -179,111 +207,128 @@ func TestRingStreamUsageGatherer_GetStreamUsage(t *testing.T) {
 
 func TestRingStreamUsageGatherer_GetPartitionConsumers(t *testing.T) {
 	tests := []struct {
-		name                              string
-		instances                         []ring.InstanceDesc
-		expectedAssignedPartitionsRequest []*logproto.GetAssignedPartitionsRequest
-		getAssignedPartitionsResponses    []*logproto.GetAssignedPartitionsResponse
-		getAssignedPartitionsResponseErrs []error
-		expected                          map[int32]string
+		name string
+		// Instances contains the complete set of instances that should be mocked.
+		// For example, if a test case is expected to make RPC calls to one instance,
+		// then just one InstanceDesc is required.
+		instances []ring.InstanceDesc
+		// The size of the following slices must match len(instances), where each
+		// value contains the expected request/response for the instance at the
+		// same index in the instances slice. If a request/response is not expected,
+		// the value can be set to nil.
+		expectedAssignedPartitionsRequests []*logproto.GetAssignedPartitionsRequest
+		getAssignedPartitionsResponses     []*logproto.GetAssignedPartitionsResponse
+		getAssignedPartitionsResponseErrs  []error
+		// The expected result.
+		expected map[int32]string
 	}{{
 		name: "single instance returns its partitions",
 		instances: []ring.InstanceDesc{{
-			Addr: "instance-1",
+			Addr: "instance-0",
```

Comment:
I made all the numbering consistent across tests as sometimes we started from `0` and other times started from `1`.